modules/rostests/kmtests/ntos_ob/ObQuery.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * PROJECT:     ReactOS kernel-mode tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Kernel mode tests for object information querying
+ * COPYRIGHT:   Copyright 2023 George Bișoc <george.bisoc@reactos.org>
+ */
+
+#include <kmt_test.h>
+
+#define OBJ_REGISTRY_TYPE_INFO_SIZE   104
+#define OBJ_GLOBAL_DIR_TYPE_INFO_SIZE 116
+
+static
+VOID
+ObjectBasicInformationTests(VOID)
+{
+    NTSTATUS Status;
+    HANDLE KeyHandle;
+    HANDLE GlobalDirHandle;
+    OBJECT_BASIC_INFORMATION BasicInfo;
+    ULONG ReturnLength;
+    OBJECT_ATTRIBUTES ObjectAttributes;
+    static UNICODE_STRING GlobalDir = RTL_CONSTANT_STRING(L"\\??");
+    static UNICODE_STRING RegRootKey = RTL_CONSTANT_STRING(L"\\Registry");
+
+    /* We must be in PASSIVE_LEVEL to do all of this stuff */
+    ok_irql(PASSIVE_LEVEL);
+
+    /* Create a key path to the registry root */
+    InitializeObjectAttributes(&ObjectAttributes,
+                               &RegRootKey,
+                               OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
+                               NULL,
+                               NULL);
+    Status = ZwOpenKey(&KeyHandle, KEY_READ, &ObjectAttributes);
+    if (!NT_SUCCESS(Status))
+    {
+        ok(FALSE, "Failed to open \\Registry (Status 0x%lx)\n", Status);
+        return;
+    }
+
+    /* Give 0 as information length, this must fail */
+    Status = ZwQueryObject(KeyHandle,
+                           ObjectBasicInformation,
+                           &BasicInfo,
+                           0,
+                           &ReturnLength);
+    ok_eq_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* Do a proper query now */
+    Status = ZwQueryObject(KeyHandle,
+                           ObjectBasicInformation,
+                           &BasicInfo,
+                           sizeof(BasicInfo),
+                           &ReturnLength);
+    ok_eq_hex(Status, STATUS_SUCCESS);
+
+    /* These must not be 0 because the registry is currently used */
+    ok(BasicInfo.HandleCount != 0, "\\Registry is in use but HandleCount is 0!\n");
+    ok(BasicInfo.PointerCount != 0, "\\Registry is in use but PointerCount is 0!\n");
+
+    /*
+     * NameInfoSize must be 0 because \Registry does not have
+     * directories, instead it has root keys (aka HKEYs).
+     * TypeInfoSize is fixed because it is defined by the
+     * size of an object type though, the size of a registry
+     * key object is 104.
```

Comment:
Isn't that different on x64?
Also, should that size correlate with the size of the data retrieved with ObjectTypeInformation ?

Reply:
The length of a name type is arch dependent, so whoops. Also yeah, `TypeInfoSize` correlates with `ObjectTypeInformation` class in one way or another. Just read the article by Geoff I linked.